nixos/modules/services/web-apps/hledger-web.nix

```diff
@@ -0,0 +1,77 @@
+{ lib, pkgs, config, ... }:
+with lib;                      
+let
+  cfg = config.services.hledger-web;
+in {
+  options.services.hledger-web = {
+
+    enable = mkEnableOption "hledger-web service";
+
+    serveApi = mkEnableOption "Serve only the JSON web API, without the web UI.";
+
+    host = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+        Address to listen on.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 5000;
+      example = "80";
+      description = ''
+        Port to listen on.
+      '';
+    };
+
+    capabilities = mkOption {
+      type = types.commas;
+      default = "view,add";
+      description = ''
+        Enable the view, add, and/or manage capabilities.
+      '';
+    };
+
+    file = mkOption {
+      type = types.path;
+      example = "~/.hledger.journal";
```

Comment:
Using `~` in such a path is probably a bad idea. I would not advertise that as an example.

Reply:
changed it to /home/hledger